internal/infrastructure/logger/meta_handler.go

```diff
@@ -0,0 +1,65 @@
+package logger
+
+import (
+	"context"
+
+	"golang.org/x/exp/slog"
+)
+
+type SlogMetaHandler struct {
+	handlers []slog.Handler
+}
+
+func NewSlogMetaHandler() *SlogMetaHandler {
+	return &SlogMetaHandler{
+		handlers: []slog.Handler{},
+	}
+}
+
+// Add a new handler to the collection
+func (h *SlogMetaHandler) Add(handler slog.Handler) {
+	h.handlers = append(h.handlers, handler)
+}
+
+// Enabled reports whether the handler handles records at the given level.
+// The handler ignores records whose level is lower.
+func (h *SlogMetaHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	var ret = true
+	for i, _ := range h.handlers {
+		if ret2 := h.handlers[i].Enabled(ctx, level); !ret2 {
+			ret = ret2
+		}
+	}
+	return ret
+}
+
+// WithAttrs returns a new TextHandler whose attributes consists
+// of h's attributes followed by attrs.
+func (h *SlogMetaHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	var newMeta = &SlogMetaHandler{handlers: []slog.Handler{}}
+	for i := range h.handlers {
+		h.handlers[i] = h.handlers[i].WithAttrs(attrs)
```

Comment:
Shouldn't this append the new handler to newMeta instead of modifying the current meta handler?

Reply:
Yep, fixed it. Thanks!

Reply:
+1

I like your approach with `make()`. It's more efficient than appending each handler.

Reply:
Yes, good catch. Fixed :-)